### ISSUE TITLE
[🐛 Bug]: Driver Manager downloads the chromedriver on all running instances

### ISSUE THREAD

**Issue by dennisdesutter91** [webdriverio/webdriverio#10861]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.14.4

### Node.js Version

20.5.0

### Mode

Standalone Mode

### What happened?

When running your tests on multiple instances on chrome, I saw in the logs that it was downloading the latest version of chrome for all running instances at the same time. Presumably this caused a conflict and my tests did not start. As seen in the logs: Error: spawn /tmp/chromedriver/linux-115.0.5790.170/chromedriver-linux64/chromedriver ENOENT it can't find the downloaded driver.

Lowering the maxinstances to 1 fixes the issue.

### What is your expected behavior?

It should only be downloaded once.

### How to reproduce the bug.

Run multiple instances on chrome. 

### Relevant log output

```typescript
[0-1] RUNNING in chrome - file:///src/test/specs/login.spec.js, file:///src/test/specs/Content%20Types%20CRU/newsArticle.spec.js
[0-1] 2023-08-04T08:20:23.639Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] RUNNING in chrome - file:///src/test/specs/dataLayer.spec.js
[0-0] 2023-08-04T08:20:23.765Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2023-08-04T08:20:24.109Z INFO webdriver: Setting up Chrome v115.0.5790.170
[0-1] 2023-08-04T08:20:24.109Z INFO webdriver: Setting up Chrome v115.0.5790.170
[0-1] 2023-08-04T08:20:24.174Z INFO webdriver: Downloading Chrome 0.01%
[0-0] 2023-08-04T08:20:24.200Z INFO webdriver: Downloading Chrome 0.01%
[0-0] 2023-08-04T08:20:25.201Z INFO webdriver: Downloading Chrome 43.60%
[0-1] 2023-08-04T08:20:25.223Z INFO webdriver: Downloading Chrome 43.17%
[0-0] 2023-08-04T08:20:26.201Z INFO webdriver: Downloading Chrome 90.79%
[0-1] 2023-08-04T08:20:26.273Z INFO webdriver: Downloading Chrome 84.83%
[0-0] 2023-08-04T08:20:30.347Z INFO webdriver: Downloading Chromedriver v115.0.5790.170
[0-0] 2023-08-04T08:20:30.431Z INFO webdriver: Downloading Chromedriver 0.22%
[0-1] 2023-08-04T08:20:30.501Z INFO webdriver: Downloading Chromedriver v115.0.5790.170
[0-1] 2023-08-04T08:20:30.544Z INFO webdriver: Downloading Chromedriver 0.22%
[0-0] 2023-08-04T08:20:30.548Z ERROR @wdio/runner: Error: unexpected EOF
[0-0]     at /builds/mediahuis-test-automation/cue-team/mediahuis-test-automation-cue-e2e/node_modules/yauzl/index.js:629:23
[0-0]     at /builds/mediahuis-test-automation/cue-team/mediahuis-test-automation-cue-e2e/node_modules/fd-slicer/index.js:32:7
[0-0]     at FSReqCallback.wrapper [as oncomplete] (node:fs:701:5)
[0-1] 2023-08-04T08:20:30.606Z ERROR @wdio/runner: Error: ENOENT: no such file or directory, open '/tmp/chromedriver/chromedriver-linux64.zip'
[0-0] FAILED in chrome - file:///src/test/specs/dataLayer.spec.js
2023-08-04T08:20:30.689Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-08-04T08:20:30.689Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-08-04T08:20:30.690Z INFO @wdio/local-runner: Start worker 0-2 with arg: run,./wdio.conf.js,--suite,regression
[0-1] FAILED in chrome - file:///src/test/specs/login.spec.js, file:///src/test/specs/Content%20Types%20CRU/newsArticle.spec.js
2023-08-04T08:20:30.740Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-08-04T08:20:30.740Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-08-04T08:20:30.742Z INFO @wdio/local-runner: Start worker 0-3 with arg: run,./wdio.conf.js,--suite,regression
[0-3] 2023-08-04T08:20:31.700Z INFO @wdio/local-runner: Run worker command: run
[0-2] 2023-08-04T08:20:31.827Z INFO @wdio/local-runner: Run worker command: run
[0-3] RUNNING in chrome - file:///src/test/specs/login.spec.js, file:///src/test/specs/Content%20Types%20CRU/opinionArticle.spec.js
[0-3] 2023-08-04T08:20:32.350Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-2] RUNNING in chrome - file:///src/test/specs/login.spec.js, file:///src/test/specs/Content%20Types%20CRU/featureArticle.spec.js
[0-2] 2023-08-04T08:20:32.516Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-3] 2023-08-04T08:20:32.573Z INFO webdriver: Setting up Chrome v115.0.5790.170
[0-3] 2023-08-04T08:20:32.575Z INFO webdriver: Downloading Chromedriver v115.0.5790.170
[0-3] node:events:492
[0-3]       throw er; // Unhandled 'error' event
[0-3]       ^
[0-3] 
[0-3] Error: spawn /tmp/chromedriver/linux-115.0.5790.170/chromedriver-linux64/chromedriver ENOENT
[0-3]     at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
[0-3]     at onErrorNT (node:internal/child_process:484:16)
[0-3]     at processTicksAndRejections (node:internal/process/task_queues:82:21)
[0-3] Emitted 'error' event on ChildProcess instance at:
[0-3]     at Process.ChildProcess._handle.onexit (node:internal/child_process:292:12)
[0-3]     at onErrorNT (node:internal/child_process:484:16)
[0-3]     at processTicksAndRejections (node:internal/process/task_queues:82:21) {
[0-3]   errno: -2,
[0-3]   code: 'ENOENT',
[0-3]   syscall: 'spawn /tmp/chromedriver/linux-115.0.5790.170/chromedriver-linux64/chromedriver',
[0-3]   path: '/tmp/chromedriver/linux-115.0.5790.170/chromedriver-linux64/chromedriver',
[0-3]   spawnargs: [ '--port=44755', '--allowed-origins=*', '--allowed-ips=' ]
[0-3] }
[0-3]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!



We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by taomoh** [comment]
We've observed similar symptoms. The difference is that setting `maxInstances` to 1 doesn't fix the problem for us; also, the execution gets stuck at this line: `webdriver: Downloading Chromedriver 0.22%`
Bizarrely, the value 0.22% re-occurs in all executions

```
wdio run ./tests/wdio.conf.js

Execution of 1 workers started at 2023-08-08T15:19:10.665Z

2023-08-08T15:19:10.679Z INFO @wdio/cli:launcher: Run onPrepare hook
2023-08-08T15:19:10.682Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-08-08T15:19:10.683Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./tests/wdio.conf.js
2023-08-08T15:19:10.697Z DEBUG @wdio/local-runner: Send command run to worker with cid "0-0"
[0-0] 2023-08-08T15:19:11.487Z INFO @wdio/local-runner: Run worker command: run
[0-0] 2023-08-08T15:19:11.656Z DEBUG @wdio/runner: init remote session
[0-0] RUNNING in chrome - file:///tests/internal/specs/billing/overview.js
[0-0] 2023-08-08T15:19:11.730Z DEBUG @wdio/runner: init remote session
[0-0] 2023-08-08T15:19:11.731Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2023-08-08T15:19:12.038Z INFO webdriver: Setting up Chrome v115.0.5790.170
[0-0] 2023-08-08T15:19:12.484Z INFO webdriver: Downloading Chrome 17.85%
[0-0] 2023-08-08T15:19:13.505Z INFO webdriver: Downloading Chrome 59.54%
[0-0] 2023-08-08T15:19:14.506Z INFO webdriver: Downloading Chrome 99.41%
[0-0] 2023-08-08T15:19:17.155Z INFO webdriver: Downloading Chromedriver v115.0.5790.170
[0-0] 2023-08-08T15:19:17.209Z INFO webdriver: Downloading Chromedriver 0.22%
```

**Comment by nextlevelbeard** [comment]
Also seeing this, but only for the first time I'm running.

With `maxInstances` = 4, 1 passed, 3 failed.

Passed log, 0-3:
```js
DEBUG @wdio/runner: init remote session
INFO webdriver: Initiate new session using the WebDriver protocol
INFO webdriver: Setting up Chrome v116.0.5845.82
INFO webdriver: Downloading Chrome 0.01%
INFO webdriver: Downloading Chrome 1.07%
INFO webdriver: Downloading Chrome 1.96%
INFO webdriver: Downloading Chrome 2.81%
INFO webdriver: Downloading Chrome 3.99%
INFO webdriver: Downloading Chrome 4.90%
INFO webdriver: Downloading Chrome 5.76%
INFO webdriver: Downloading Chrome 6.71%
INFO webdriver: Downloading Chrome 7.32%
INFO webdriver: Downloading Chrome 8.09%
INFO webdriver: Downloading Chrome 8.77%
INFO webdriver: Downloading Chrome 9.86%
INFO webdriver: Downloading Chrome 10.80%
INFO webdriver: Downloading Chrome 11.93%
INFO webdriver: Downloading Chrome 13.25%
INFO webdriver: Downloading Chrome 14.56%
INFO webdriver: Downloading Chrome 16.05%
INFO webdriver: Downloading Chrome 17.24%
INFO webdriver: Downloading Chrome 18.58%
INFO webdriver: Downloading Chrome 20.18%
INFO webdriver: Downloading Chrome 21.66%
INFO webdriver: Downloading Chrome 22.98%
INFO webdriver: Downloading Chrome 24.22%
INFO webdriver: Downloading Chrome 25.71%
INFO webdriver: Downloading Chrome 27.11%
INFO webdriver: Downloading Chrome 29.06%
INFO webdriver: Downloading Chrome 30.76%
INFO webdriver: Downloading Chrome 32.41%
INFO webdriver: Downloading Chrome 33.93%
INFO webdriver: Downloading Chrome 35.70%
INFO webdriver: Downloading Chrome 37.02%
INFO webdriver: Downloading Chrome 38.37%
INFO webdriver: Downloading Chrome 39.73%
INFO webdriver: Downloading Chrome 41.25%
INFO webdriver: Downloading Chrome 42.56%
INFO webdriver: Downloading Chrome 43.99%
INFO webdriver: Downloading Chrome 45.20%
INFO webdriver: Downloading Chrome 46.59%
INFO webdriver: Downloading Chrome 48.10%
INFO webdriver: Downloading Chrome 49.73%
INFO webdriver: Downloading Chrome 51.33%
INFO webdriver: Downloading Chrome 52.76%
INFO webdriver: Downloading Chrome 54.13%
INFO webdriver: Downloading Chrome 55.72%
INFO webdriver: Downloading Chrome 57.40%
INFO webdriver: Downloading Chrome 59.20%
INFO webdriver: Downloading Chrome 60.78%
INFO webdriver: Downloading Chrome 63.03%
INFO webdriver: Downloading Chrome 65.13%
INFO webdriver: Downloading Chrome 67.11%
INFO webdriver: Downloading Chrome 69.32%
INFO webdriver: Downloading Chrome 71.13%
INFO webdriver: Downloading Chrome 72.51%
INFO webdriver: Downloading Chrome 74.34%
INFO webdriver: Downloading Chrome 76.06%
INFO webdriver: Downloading Chrome 77.93%
INFO webdriver: Downloading Chrome 79.75%
INFO webdriver: Downloading Chrome 81.38%
INFO webdriver: Downloading Chrome 83.39%
INFO webdriver: Downloading Chrome 85.61%
INFO webdriver: Downloading Chrome 87.19%
INFO webdriver: Downloading Chrome 88.96%
INFO webdriver: Downloading Chrome 90.71%
INFO webdriver: Downloading Chrome 92.69%
INFO webdriver: Downloading Chrome 94.67%
INFO webdriver: Downloading Chrome 96.40%
INFO webdriver: Downloading Chrome 98.43%
INFO webdriver: Downloading Chromedriver v116.0.5845.82
INFO webdriver: Downloading Chromedriver 0.19%
INFO webdriver: Downloading Chromedriver 19.41%
INFO webdriver: Downloading Chromedriver 38.04%
INFO webdriver: Downloading Chromedriver 88.50%
INFO webdriver: Started ChromeDriver v116.0.5845.82 with params --port=53014 --allowed-origins=* --allowed-ips= in 75736ms on port 53014
```

Failed instances, 0-0, 0-1, 0-2
```js

DEBUG @wdio/runner: init remote session
INFO webdriver: Initiate new session using the WebDriver protocol
INFO webdriver: Setting up Chrome v116.0.5845.82
INFO webdriver: Downloading Chrome 0.01%
INFO webdriver: Downloading Chrome 1.23%
INFO webdriver: Downloading Chrome 2.23%
INFO webdriver: Downloading Chrome 3.22%
INFO webdriver: Downloading Chrome 4.51%
INFO webdriver: Downloading Chrome 5.46%
INFO webdriver: Downloading Chrome 6.45%
INFO webdriver: Downloading Chrome 7.43%
INFO webdriver: Downloading Chrome 8.07%
INFO webdriver: Downloading Chrome 8.97%
INFO webdriver: Downloading Chrome 9.71%
INFO webdriver: Downloading Chrome 10.65%
INFO webdriver: Downloading Chrome 11.71%
INFO webdriver: Downloading Chrome 12.88%
INFO webdriver: Downloading Chrome 14.14%
INFO webdriver: Downloading Chrome 15.38%
INFO webdriver: Downloading Chrome 16.38%
INFO webdriver: Downloading Chrome 17.34%
INFO webdriver: Downloading Chrome 18.34%
INFO webdriver: Downloading Chrome 19.65%
INFO webdriver: Downloading Chrome 20.90%
INFO webdriver: Downloading Chrome 22.24%
INFO webdriver: Downloading Chrome 22.90%
INFO webdriver: Downloading Chrome 23.56%
INFO webdriver: Downloading Chrome 24.21%
INFO webdriver: Downloading Chrome 25.03%
INFO webdriver: Downloading Chrome 25.91%
INFO webdriver: Downloading Chrome 26.76%
INFO webdriver: Downloading Chrome 27.79%
INFO webdriver: Downloading Chrome 29.11%
INFO webdriver: Downloading Chrome 30.21%
INFO webdriver: Downloading Chrome 31.26%
INFO webdriver: Downloading Chrome 32.33%
INFO webdriver: Downloading Chrome 33.47%
INFO webdriver: Downloading Chrome 34.41%
INFO webdriver: Downloading Chrome 35.50%
INFO webdriver: Downloading Chrome 36.48%
INFO webdriver: Downloading Chrome 37.76%
INFO webdriver: Downloading Chrome 39.05%
INFO webdriver: Downloading Chrome 40.47%
INFO webdriver: Downloading Chrome 41.95%
INFO webdriver: Downloading Chrome 43.18%
INFO webdriver: Downloading Chrome 44.28%
INFO webdriver: Downloading Chrome 45.86%
INFO webdriver: Downloading Chrome 47.48%
INFO webdriver: Downloading Chrome 48.90%
INFO webdriver: Downloading Chrome 50.08%
INFO webdriver: Downloading Chrome 51.48%
INFO webdriver: Downloading Chrome 52.87%
INFO webdriver: Downloading Chrome 54.32%
INFO webdriver: Downloading Chrome 55.84%
INFO webdriver: Downloading Chrome 56.99%
INFO webdriver: Downloading Chrome 57.95%
INFO webdriver: Downloading Chrome 59.20%
INFO webdriver: Downloading Chrome 60.36%
INFO webdriver: Downloading Chrome 61.56%
INFO webdriver: Downloading Chrome 62.80%
INFO webdriver: Downloading Chrome 63.77%
INFO webdriver: Downloading Chrome 64.86%
INFO webdriver: Downloading Chrome 66.11%
INFO webdriver: Downloading Chrome 67.02%
INFO webdriver: Downloading Chrome 68.27%
INFO webdriver: Downloading Chrome 69.51%
INFO webdriver: Downloading Chrome 70.92%
INFO webdriver: Downloading Chrome 72.28%
INFO webdriver: Downloading Chrome 73.50%
INFO webdriver: Downloading Chrome 74.84%
INFO webdriver: Downloading Chrome 75.98%
INFO webdriver: Downloading Chrome 77.96%
INFO webdriver: Downloading Chrome 80.09%
INFO webdriver: Downloading Chrome 82.25%
INFO webdriver: Downloading Chrome 83.80%
INFO webdriver: Downloading Chrome 84.71%
INFO webdriver: Downloading Chrome 85.34%
INFO webdriver: Downloading Chrome 86.28%
INFO webdriver: Downloading Chrome 87.93%
INFO webdriver: Downloading Chrome 89.86%
INFO webdriver: Downloading Chrome 91.36%
INFO webdriver: Downloading Chrome 93.13%
INFO webdriver: Downloading Chrome 94.69%
INFO webdriver: Downloading Chrome 96.18%
INFO webdriver: Downloading Chrome 97.65%
INFO webdriver: Downloading Chrome 99.28%
ERROR @wdio/runner: Error: ENOENT: no such file or directory, open '/var/folders/sx/5p7dvbm54g155b9l4ddy7ln40000gn/T/chrome/116.0.5845.82-chrome-mac-arm64.zip'

```

**Comment by christian-bromann** [comment]
@nextlevelbeard @dennisdesutter91 @taomoh I pushed a fixed that makes the testrunner setup everything before only once. This should stabilise everything and should make execution faster.

**Comment by taomoh** [comment]
For future reference, if you're running tests on a Docker image: node:18 image seems to be problematic. Switching to another image (e.g. ghcr.io/puppeteer/puppeteer:21.0.3) fixes the problem

**Comment by tgomez92** [comment]
Hey, Im having the same issue as the post. Im using node 16, the weird thing is that in the past the test were working correctly. Here is the error:
`
Execution of 1 workers started at 2023-10-03T18:34:36.284Z

2023-10-03T18:34:36.291Z INFO @wdio/cli:launcher: Run onPrepare hook
2023-10-03T18:34:36.292Z INFO @wdio/utils: Setting up browser driver for: chrome@stable
2023-10-03T18:34:36.292Z INFO @wdio/utils: Setting up browser binaries for: chrome@stable
2023-10-03T18:34:36.343Z INFO webdriver: Downloading Chromedriver v117.0.5938.132
2023-10-03T18:34:37.242Z WARN webdriver: Couldn't download Chromedriver v117.0.5938.132: Download failed: server returned code 404. URL: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/117.0.5938.132/mac-arm64/chromedriver-mac-arm64.zip, trying to find known good version...
2023-10-03T18:34:37.852Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-10-03T18:34:37.854Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.js
[0-0] 2023-10-03T18:34:38.248Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in chrome - file:///test/specs/loginPage.steps.js
[0-0] 2023-10-03T18:34:38.618Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2023-10-03T18:34:38.652Z INFO webdriver: Downloading Chromedriver v117.0.5938.132
[0-0] 2023-10-03T18:34:39.613Z WARN webdriver: Couldn't download Chromedriver v117.0.5938.132: Download failed: server returned code 404. URL: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/117.0.5938.132/mac-arm64/chromedriver-mac-arm64.zip, trying to find known good version...
[0-0] node:events:368
[0-0]       throw er; // Unhandled 'error' event
[0-0]       ^
[0-0] 
[0-0] Error: spawn /var/folders/g6/kj27xf_x1jvbq6t9_tgtp9gw0000gn/T/chromedriver/mac_arm-117.0.5938.92/chromedriver-mac-arm64/chromedriver ENOENT
[0-0]     at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
[0-0]     at onErrorNT (node:internal/child_process:477:16)
[0-0]     at processTicksAndRejections (node:internal/process/task_queues:83:21)
[0-0] Emitted 'error' event on ChildProcess instance at:
[0-0]     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
[0-0]     at onErrorNT (node:internal/child_process:477:16)
[0-0]     at processTicksAndRejections (node:internal/process/task_queues:83:21) {
[0-0]   errno: -2,
[0-0]   code: 'ENOENT',
[0-0]   syscall: 'spawn /var/folders/g6/kj27xf_x1jvbq6t9_tgtp9gw0000gn/T/chromedriver/mac_arm-117.0.5938.92/chromedriver-mac-arm64/chromedriver',
[0-0]   path: '/var/folders/g6/kj27xf_x1jvbq6t9_tgtp9gw0000gn/T/chromedriver/mac_arm-117.0.5938.92/chromedriver-mac-arm64/chromedriver',
[0-0]   spawnargs: [ '--port=60374', '--allowed-origins=*', '--allowed-ips=0.0.0.0' ]
[0-0] }
[0-0] FAILED in chrome - file:///test/specs/loginPage.steps.js
2023-10-03T18:34:39.709Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-10-03T18:34:39.710Z INFO @wdio/cli:launcher: Run onComplete hook`

Anyone has an idea how to fix this problem? Thanks!

**Comment by th0car** [comment]
I receive a more or less identical error as @tgomez92, but for me it shows after a _successful_ download of chromedriver is logged. Another difference is that my node:events code is 491 instead of 368.
This happens when trying to execute test for a wdio project in a Github Actions Workflow (**rhel** image is used) 
Wdio 8.27.1, Node 20.

**Comment by haywirez** [comment]
Downgrading fixed the issue for me (in my case, from `8.31.1` to `8.12.1`) 🤔

**Comment by ammarahnizam** [comment]
Clearing browser cache and relaunching browser Chrome resolved this issue
